test/cmd/images.sh

```diff
@@ -20,125 +21,126 @@ USE_IMAGES=${USE_IMAGES:-$defaultimage}
-[ -z "$(oc get imageStreams postgresql --template="{{.status.dockerImageRepository}}")" ]
-[ -z "$(oc get imageStreams mongodb --template="{{.status.dockerImageRepository}}")" ]
-[ -z "$(oc get imageStreams wildfly --template="{{.status.dockerImageRepository}}")" ]
-tryuntil oc get imagestreamTags mysql:latest
-[ -n "$(oc get imagestreams mysql --template="{{ index .metadata.annotations \"openshift.io/image.dockerRepositoryCheck\"}}")" ]
-oc describe istag/mysql:latest
-[ "$(oc describe istag/mysql:latest | grep "Environment:")" ]
-[ "$(oc describe istag/mysql:latest | grep "Image Created:")" ]
-[ "$(oc describe istag/mysql:latest | grep "Image Name:")" ]
+os::cmd::try_until_success 'oc get imagestreamtags wildfly:latest'
+os::cmd::expect_success_and_text "oc get imageStreams wildfly --template='{{ index .metadata.annotations \"openshift.io/image.dockerRepositoryCheck\"}}'" '[0-9]{4}\-[0-9]{2}\-[0-9]{2}' # expect a date like YYYY-MM-DD
+os::cmd::expect_success_and_text 'oc get istag' 'wildfly'
+os::cmd::expect_success 'oc annotate istag/wildfly:latest foo=bar'
+os::cmd::expect_success_and_text 'oc get istag/wildfly:latest -o jsonpath={.metadata.annotations.foo}' 'bar'
+os::cmd::expect_success 'oc annotate istag/wildfly:latest foo-'
+os::cmd::expect_success_and_not_text 'oc get istag/wildfly:latest -o jsonpath={.metadata.annotations}' 'bar'
+os::cmd::expect_success 'oc delete imageStreams ruby'
+os::cmd::expect_success 'oc delete imageStreams nodejs'
+os::cmd::expect_success 'oc delete imageStreams wildfly'
+os::cmd::expect_success 'oc delete imageStreams postgresql'
+os::cmd::expect_success 'oc delete imageStreams mongodb'
+os::cmd::expect_failure 'oc get imageStreams ruby'
+os::cmd::expect_failure 'oc get imageStreams nodejs'
+os::cmd::expect_failure 'oc get imageStreams postgresql'
+os::cmd::expect_failure 'oc get imageStreams mongodb'
+os::cmd::expect_failure 'oc get imageStreams wildfly'
+os::cmd::try_until_success 'oc get imagestreamTags mysql:5.5'
+os::cmd::try_until_success 'oc get imagestreamTags mysql:5.6'
+os::cmd::expect_success_and_text "oc get imagestreams mysql --template='{{ index .metadata.annotations \"openshift.io/image.dockerRepositoryCheck\"}}'" '[0-9]{4}\-[0-9]{2}\-[0-9]{2}' # expect a date like YYYY-MM-DD
+os::cmd::expect_success 'oc describe istag/mysql:latest'
+os::cmd::expect_success_and_text 'oc describe istag/mysql:latest' 'Environment:'
+os::cmd::expect_success_and_text 'oc describe istag/mysql:latest' 'Image Created:'
+os::cmd::expect_success_and_text 'oc describe istag/mysql:latest' 'Image Name:'
 name=$(oc get istag/mysql:latest --template='{{ .image.metadata.name }}')
 imagename="isimage/mysql@${name:0:15}"
-oc describe "${imagename}"
-[ "$(oc describe ${imagename} | grep "Environment:")" ]
-[ "$(oc describe ${imagename} | grep "Image Created:")" ]
-[ "$(oc describe ${imagename} | grep "Image Name:")" ]
+os::cmd::expect_success 'oc describe "${imagename}"'
```

Comment:
Also every command is run in a subshell so you can't set variables with it. I think a doc would be good, actually.